### PR TITLE
Console - toggle GDPR

### PR DESCRIPTION
**Pull request by RemiDesgrange**
for example for internal use of georchestra or for country where GDPR
    doesn't apply.

See https://github.com/georchestra/datadir/pull/173 for datadir related PR

Issue : https://github.com/georchestra/georchestra/issues/2795

### CODE REVIEW

File: console/src/main/java/org/georchestra/console/ws/backoffice/users/UsersExport.java [hunk 1]
```diff
@@ -53,6 +56,10 @@
 
     private UserInfoExporter accountInfoExporter;
 
+    @VisibleForTesting
+    @Value("${gdpr.enable}")
+    String gdprEnable;
```
Comment (pmauduit) [inline console/src/main/java/org/georchestra/console/ws/backoffice/users/UsersExport.java]: the VisibleForTesting annotation might suggest more than "package-private" modifier (i.e. public or so). I'd have put the member private, and defined getters/setters on it, but no idea how spring can react to this, to "autowire" it using the value annotation.

Also isn't spring capable of automatically casting true / false to a boolean ?

File: console/src/main/java/org/georchestra/console/ws/backoffice/users/UsersExport.java [hunk 2]
```diff
@@ -68,6 +75,15 @@
     @RequestMapping(method = RequestMethod.GET, value = "/account/gdpr/download", produces = "application/zip")
     public void downloadUserData(HttpServletResponse response)
             throws NameNotFoundException, DataServiceException, IOException {
+        /*
+         * Disabling this endpoint is the gdpr.enable property is set to false. we are
```
Comment (pmauduit) [inline console/src/main/java/org/georchestra/console/ws/backoffice/users/UsersExport.java]: "*if the gdpr.enable"

File: console/src/main/java/org/georchestra/console/ws/backoffice/users/UsersExport.java [hunk 3]
```diff
@@ -68,6 +75,15 @@
...
+         * not using spring boot, so we cannot disable this endpoint with
+         * the @ConditionalOnProperty annotation.
```
Comment (pmauduit) [inline console/src/main/java/org/georchestra/console/ws/backoffice/users/UsersExport.java]: not sure it is necessary to mention the fact that the annotation is unavailable

File: console/src/main/java/org/georchestra/console/ws/backoffice/users/UsersExport.java (hunk 1 above)
Comment (pmauduit) [reply console/src/main/java/org/georchestra/console/ws/backoffice/users/UsersExport.java]: relying on the rest of your PR, spring does not care about private member variables when it deals with the @Value annotation, as it probably does some Java reflection to access the field it needs to set.

File: console/src/main/java/org/georchestra/console/ws/backoffice/users/UsersExport.java (hunk 3 above)
Comment (RemiDesgrange) [reply console/src/main/java/org/georchestra/console/ws/backoffice/users/UsersExport.java]: Yeah, dunno, I was just bored that I found this perfect annotation in the doc and then my IDE said "Nope, you can't use that" So I mentioned it, but you're probably right.

**Comment by fvanderbiest** [comment]
Customer is not interested anymore in this feature. Closing.

**Comment by fvanderbiest** [comment]
Well, it is considered helpful in some cases ...

@RemiDesgrange can you rebase ? Thanks

**Review by cmangeat — APPROVED** [review]

**Comment by pmauduit** [comment]
Runtime tested:

* [ ] UsersController: /account/gdpr/delete
* [ ] EditUserDetailsFormController: /account/userdetails
* [ ] UserExport: /account/gdpr/download

If gdpr.enable = false, then the previous endpoints are not accessible, if gdpr.enable = true, they are.

**Comment by pmauduit** [comment]
rebased / force-pushed

**Pull request merged**